You're calling `collect`, which creates a new collection, and then calling `into_iter`. However, the value you had before that was already an iterator! So, you can call....